### DISCUSSION TITLE
GraphQL many to any relations

### DISCUSSION THREAD

**Discussion by cpt-n3mo** [directus/directus#10447] [Archive]
Good day !

I was wondering if there was a particular way of getting more then a ID from a many to any relation..?
with a many to many relation `subCatagories` one can do something like this

```gql
query {
  catagories {
    id
    name
    icon
    subCatagories {
      subCatagories_id {
        name
        id
      }
    }
  }
}
```

but this does not seem to work if i change `subCatagories` to a many to any relation, i can only seem to get a ID and a collection.. is there a way to get the data directly without needing to do a second query..?

kind regards..

**Comment by rijkvanzanten** [comment]
You can use the union types in GraphQL to fetch the nested data from the related collections in a many to any. See https://docs.directus.io/reference/introduction/#many-to-any-union-types for more information.

**Reply by cpt-n3mo** [reply]
ah fast as always.. 
i will poke around some more in the morning.. for now with that way im getting a with `fields is not iterable`.. 
thx for the quick help.. !
 EDIT: Worked like a charm.. thx!

Groet uit Holland

**Reply by rijkvanzanten** [reply]
Groetjes terug!

**Reply by Scottzonn** [reply]
Can you please post your working query? I still can't work this out! thanks

**Reply by rijkvanzanten** [reply]
@Scottzonn What have you tried so far? [The query in the docs](https://docs.directus.io/reference/introduction/#graphql) is how you'd do this, but the exact naming depends on your schema / permissions
